File: modules/swagger-servlet/src/main/scala/com/wordnik/swagger/servlet/config/DefaultServletReaderConfig.scala
package com.wordnik.swagger.servlet.config

import com.wordnik.swagger.config._
import com.wordnik.swagger.reader._

import javax.servlet.ServletConfig
import javax.servlet.http.HttpServlet

class DefaultServletReaderConfig extends HttpServlet {
  override def init(servletConfig: ServletConfig) = {
    super.init(servletConfig)

    implicit val config = servletConfig
    val webConfig = new WebXMLReader
    ConfigFactory.config = webConfig

    val scanner = new ServletScanner
    ScannerFactory.setScanner(scanner)
    scanner.setResourcePackage(webConfig.resourcePackage)
    ClassReaders.reader = Some(new ServletReader )
  }
}

File: modules/swagger-servlet/src/main/java/com/wordnik/swagger/servlet/config/DefaultServletReaderConfig.java
package com.wordnik.swagger.servlet.config;

import javax.servlet.ServletConfig;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;

import com.wordnik.swagger.jaxrs.config.BeanConfig;

public class DefaultServletReaderConfig extends HttpServlet {
  @Override
  public void init(ServletConfig servletConfig) throws ServletException {
    super.init(servletConfig);

    BeanConfig bc = new BeanConfig();
    bc.setResourcePackage(servletConfig.getInitParameter("swagger.resource.package"));
    bc.setScan(true);

    System.out.println("BEEEEEEP\007");
  }
}
